hugegraph-api/src/main/java/com/baidu/hugegraph/api/graph/BatchAPI.java

```diff
@@ -71,4 +82,99 @@ public BatchAPI() {
             batchWriteThreads.decrementAndGet();
         }
     }
+
+    // TODO: Design more flexible for Element?
+    @JsonIgnoreProperties(value = {"type"})
+    protected static abstract class JsonElement implements Checkable {
+
+        @JsonProperty("label")
+        public String label;
+        @JsonProperty("properties")
+        public Map<String, Object> properties;
+        @JsonProperty("type")
+        public String type;
+
+        @Override
+        public abstract void checkCreate(boolean isBatch);
+
+        @Override
+        public abstract void checkUpdate();
+
+        protected abstract Object[] properties();
+    }
+
+    /**
+     * Get vertex/edge ID from JsonVerex/JsonEdge
+     * TODO: Should support AUTOMATIC ElementID ?
+     * TODO: How to get JsonEdge's fields ?
+     */
+    protected String getElementId(HugeGraph g, String labelId,
```

Comment:
seems can split into two methods directly: getVertexId(g, label, JsonVertex) and getEdgeId(g, label, JsonEdge).
can avoid the method too long and reduce the number of parameters.
If need, move getVertexId into VertexAPI, getEdgeId into EdgeAPI